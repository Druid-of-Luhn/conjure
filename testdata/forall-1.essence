language Essence 2.0

letting digit be domain int(0..9)

find x : digit
find y : digit

letting xy be x * y

letting myForall be quantifier {
    lambda { x:bool, y:bool -> x /\ y }
    lambda { x:bool, y:bool -> x => y }
    true
    }

such that
    x = 3*y,
    xy > 0
