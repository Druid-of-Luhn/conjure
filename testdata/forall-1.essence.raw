Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Letting
                         , "digit"
                         , DomainIntegerFromTo (
                             Just ( ValueInteger 0 ) ) (
                             Just ( ValueInteger 9 ) )
                         )
                       , ( Find
                         , "x"
                         , Identifier "digit"
                         )
                       , ( Find
                         , "y"
                         , Identifier "digit"
                         )
                       , ( Letting
                         , "xy"
                         , GenericNode
                             Times
                             [ Identifier "x"
                             , Identifier "y"
                             ]
                         )
                       , ( Letting
                         , "forall_glueOp"
                         , DeclLambda
                             [ ( "x"
                               , TypeBoolean
                               )
                             , ( "y"
                               , TypeBoolean
                               )
                             ] (
                             GenericNode
                               And
                               [ Identifier "x"
                               , Identifier "y"
                               ] )
                         )
                       , ( Letting
                         , "forall_skipOp"
                         , DeclLambda
                             [ ( "x"
                               , TypeBoolean
                               )
                             , ( "y"
                               , TypeBoolean
                               )
                             ] (
                             GenericNode
                               Imply
                               [ Identifier "x"
                               , Identifier "y"
                               ] )
                         )
                       , ( Letting
                         , "forall"
                         , DeclQuantifier (
                             Identifier "forall_glueOp" ) (
                             Identifier "forall_skipOp" ) (
                             ValueBoolean True )
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Eq
                      [ Identifier "x"
                      , GenericNode
                          Times
                          [ ValueInteger 3
                          , Identifier "y"
                          ]
                      ]
                  , GenericNode
                      Gt
                      [ Identifier "xy"
                      , ValueInteger 0
                      ]
                  ]
  }