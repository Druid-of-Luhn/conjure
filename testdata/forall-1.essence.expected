language Essence 2.0

letting digit be domain int(0..9)
find x : digit
find y : digit
letting xy be x * y
letting forall_glueOp be lambda { x : bool, y : bool -> x /\ y }
letting forall_skipOp be lambda { x : bool, y : bool -> x => y }
letting forall be quantifier { forall_glueOp
                               forall_glueOp
                               true
                             }

such that
    x = 3 * y,
    xy > 0
